src/impl_constructors.rs

```diff
@@ -171,6 +171,30 @@ where
         }
         eye
     }
+
+    /// Create a 2D matrix from its diagonal
+    ///
+    /// **Panics** if `diag.len() * diag.len()` would overflow `isize`.
+    ///
+    /// ```rust
+    /// use ndarray::{Array2, arr1, arr2};
+    ///
+    /// # #[cfg(feature = "approx")] {
```

Comment:
Is this attribute necessary? I think it should be possible to remove it.

Reply:
@jturner314 if you meant `cfg`, approx is an optional dependency, in `Cargo.toml` as far as I can tell,
```
approx = { version = "0.3.2", optional = true }
```
and that's what other examples are doing?

Reply:
Sorry, I wasn't clear. You're correct that `approx` is an optional dependency/feature, and some of the other examples have a `#[cfg(feature = "approx")]` annotation because they use the `approx` feature. In this case, the example isn't using `approx`, so the lines `    /// # #[cfg(feature = "approx")] {` and `    /// # }` shouldn't be necessary. (You should be able to remove them and have `cargo test` still pass.)

Reply:
Aww, yes, sorry, I forgot I was no longer using float equality tests there. Thanks for your patience. Should be fixed now.